### PR TITLE
[#165] /logicalpath: Add POST operation for mkdir

### PR DESCRIPTION
**Pull request by alanking**
Adds a POST function for the /logicalpath endpoint which creates a
collection a la imkdir. If the collection parameter is not used, this
operation does nothing.

### CODE REVIEW

**Comment by alanking** [comment]
I ignored clang-format for the "generated" pieces of code, as we have been doing.

File: irods_specific_implementation/irods_rest_logical_path_api_implementation.h [hunk 1]
```diff
@@ -246,6 +281,37 @@ namespace irods::rest
...
+        {
+            const auto create_parent_collections =
+                "1" == _request.query().get("create-parent-collections").getOrElse("0");
+
+            auto conn = get_connection(_request.headers().getRaw("authorization").value());
+
+            try {
+                if (create_parent_collections) {
+                    if (!fscli::create_collections(*conn(), _path)) {
+                        info("Collection [{}] exists. Nothing to do.", _path.c_str());
```
Comment (trel) [inline irods_specific_implementation/irods_rest_logical_path_api_implementation.h:295]: possible "info" -> "debug" ?

File: packaging/test_irods_client_rest_cpp.py [hunk 2]
```diff
@@ -932,3 +932,73 @@ def test_trim_collection(self):
                 admin.run_icommand(['irm', '-r', '-f', coll_name])
                 for resc in resources:
                    admin.run_icommand(['iadmin', 'rmresc', resc])
+
+
+def collection_exists(session, full_collection_path):
+    out = session.run_icommand(['iquest',
```
Comment (trel) [inline packaging/test_irods_client_rest_cpp.py:938]: `run_icommand` returns a tuple?

File: irods_specific_implementation/irods_rest_logical_path_api_implementation.h (hunk 1 above)
Comment (alanking) [reply irods_specific_implementation/irods_rest_logical_path_api_implementation.h:295]: Sure thing.

File: packaging/test_irods_client_rest_cpp.py (hunk 2 above)
Comment (alanking) [reply packaging/test_irods_client_rest_cpp.py:938]: Yeah, notice the `[0]` at the end of the line (well, I guess it's on the next line, 939).

Comment (trel) [reply packaging/test_irods_client_rest_cpp.py:938]: oh, yeah, totally saw that.

File: irods_specific_implementation/irods_rest_logical_path_api_implementation.h [hunk 3]
```diff
@@ -123,7 +123,42 @@ namespace irods::rest
...
+            try {
+                const auto logical_path = _request.query().get("logical-path").get();
+                if (logical_path.empty()) {
+                    return make_error_response(SYS_INVALID_INPUT_PARAM, "logical-path parameter is required");
+                }
+                const fs::path path = decode_url(logical_path);
+
+                const auto create_collection = "1" == _request.query().get("collection").getOrElse("0");
+                if (create_collection) {
+                    return mkdir(path.c_str(), _request);
```
Comment (alanking) [inline irods_specific_implementation/irods_rest_logical_path_api_implementation.h:140]: path.c_str() -> path

**Comment by trel** [comment]
are we good here?   please to squash when happy.

**Comment by alanking** [comment]
I think so. Squashed

**Comment by trel** [comment]
okay, so the 'touch' functionality will come a little later.   yes, please # if this feels done.

**Comment by alanking** [comment]
I have it implemented, but it would definitely need a discussion or three: https://github.com/irods/irods_client_rest_cpp/issues/168

Added #

**Pull request merged**